From bd9c4bc383c7c18a8a7ca7bd991e6d38edc9d0c3 Mon Sep 17 00:00:00 2001
From: Kenvish Butani <kbutani@qti.qualcomm.com>
Date: Wed, 13 Aug 2025 17:30:06 +0530
Subject: [PATCH] UPSTREAM: DP: wifi: ath12k: Reordered Tx Enqueue Drop Enum
 fields

Reordered Tx Enqueue Drop Enum fields for efficient usage in
per packet path

Patch-dependency: None

Signed-off-by: Kenvish Butani <kbutani@qti.qualcomm.com>
---
 drivers/net/wireless/ath/ath12k/dp_stats.h | 12 ++++++------
 drivers/net/wireless/ath/ath12k/wifi7/hw.c |  6 +++---
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp_stats.h b/drivers/net/wireless/ath/ath12k/dp_stats.h
index eab4c13..1c26e92 100644
--- a/drivers/net/wireless/ath/ath12k/dp_stats.h
+++ b/drivers/net/wireless/ath/ath12k/dp_stats.h
@@ -34,6 +34,12 @@ enum ath12k_stats_object {
 
 enum ath12k_dp_tx_enq_error {
 	DP_TX_ENQ_SUCCESS = 0,
+	DP_TX_ENQ_DROP_TCL_DESC_NA,
+	DP_TX_ENQ_DROP_SW_DESC_NA,
+	DP_TX_ENQ_DROP_MAX_TX_LIMIT_FAST,
+	DP_TX_ENQ_DROP_INV_ARVIF_FAST,
+	DP_TX_ENQ_DROP_INV_PDEV_FAST,
+	DP_TX_ENQ_DROP_INV_ENCAP_FAST,
 	DP_TX_ENQ_DROP_MISC,
 	DP_TX_ENQ_DROP_VIF_TYPE_MON,
 	DP_TX_ENQ_DROP_INV_LINK,
@@ -44,18 +50,12 @@ enum ath12k_dp_tx_enq_error {
 	DP_TX_ENQ_DROP_INV_PEER,
 	DP_TX_ENQ_DROP_CRASH_FLUSH,
 	DP_TX_ENQ_DROP_NON_DATA_FRAME,
-	DP_TX_ENQ_DROP_SW_DESC_NA,
 	DP_TX_ENQ_DROP_ENCAP_RAW,
 	DP_TX_ENQ_DROP_ENCAP_802_3,
 	DP_TX_ENQ_DROP_DMA_ERR,
 	DP_TX_ENQ_DROP_EXT_DESC_NA,
 	DP_TX_ENQ_DROP_HTT_MDATA_ERR,
-	DP_TX_ENQ_DROP_TCL_DESC_NA,
 	DP_TX_ENQ_TCL_DESC_RETRY,
-	DP_TX_ENQ_DROP_INV_ARVIF_FAST,
-	DP_TX_ENQ_DROP_INV_PDEV_FAST,
-	DP_TX_ENQ_DROP_MAX_TX_LIMIT_FAST,
-	DP_TX_ENQ_DROP_INV_ENCAP_FAST,
 	DP_TX_ENQ_DROP_BRIDGE_VDEV,
 	DP_TX_ENQ_DROP_ARSTA_NA,
 	DP_TX_ENQ_ERR_MAX,
diff --git a/drivers/net/wireless/ath/ath12k/wifi7/hw.c b/drivers/net/wireless/ath/ath12k/wifi7/hw.c
index 508766c..bbe3dc5 100644
--- a/drivers/net/wireless/ath/ath12k/wifi7/hw.c
+++ b/drivers/net/wireless/ath/ath12k/wifi7/hw.c
@@ -1368,7 +1368,7 @@ static void ath12k_wifi7_mac_op_tx(struct ieee80211_hw *hw,
 		if (!arvif || !arvif->ar) {
 			ath12k_wifi7_ieee80211_free_txskb(hw, skb, dp_vif,
 							  DP_TX_ENQ_DROP_INV_ARVIF_FAST,
-							  ring_id, false);
+							  ring_id, true);
 
 			return;
 		}
@@ -1379,7 +1379,7 @@ static void ath12k_wifi7_mac_op_tx(struct ieee80211_hw *hw,
 		if (!dp_pdev) {
 			ath12k_wifi7_ieee80211_free_txskb(hw, skb, dp_vif,
 							  DP_TX_ENQ_DROP_INV_PDEV_FAST,
-							  ring_id, false);
+							  ring_id, true);
 
 			return;
 		}
@@ -1391,7 +1391,7 @@ static void ath12k_wifi7_mac_op_tx(struct ieee80211_hw *hw,
 				   ar->pdev_idx);
 			ath12k_wifi7_ieee80211_free_txskb(hw, skb, dp_vif,
 							  DP_TX_ENQ_DROP_MAX_TX_LIMIT_FAST,
-							  ring_id, false);
+							  ring_id, true);
 
 			return;
 		}
-- 
2.34.1

